app/views/site/offline.html.erb

```diff
@@ -1,7 +1,9 @@
+<div class="alert alert-danger text-center">
```

Comment:
For both the offline and edit pages, we end up with "alert"-style banners, but these don't match the flash banners used on the rest of the site, namely:

* They don't stretch across the whole page
* They don't have the icon alongside the message

It would be nice to make them completely consistent, rather than partially consistent. So for example, I would consider reworking this page to use the existing flash messages system to render the alert, for example by using `flash.now[:warning] = t("layouts.osm_offline")` in the controller. That way any changes made to flash messages elsewhere (e.g. adjusting margins or paddings) will apply automatically to these offline alert messages.

Reply:
I'm not exactly sure what exactly needs to be done here. Would this work or the changes need to be made somewhere else?

```
<% if Settings.status == "database_offline" %>
<p><%= flash.now[:warning] = t("layouts.osm_offline") %>
</p>
<% else %>
<p><%= flash.now[:warning] = t("layouts.osm_read_only") %>
</p>
<% end %>
```

Reply:
> Would this work

Something similar to this actually does work - the trick is that you can add flash messages from a view (since it appears the view is processed before the layout) but you need to avoid also writing the message at the same time. So no `<p>` tags and no `<%=` outputs. This works:

```
<% if Settings.status == "database_offline" %>
  <% flash.now[:warning] = t("layouts.osm_offline") %>
<% else %>
  <% flash.now[:warning] = t("layouts.osm_read_only") %>
<% end %>
```

However, I find that a bit unintuitive, since it depends on a timing issue with rendering views vs layouts, and usually flash messages are set in the controller before rendering starts. So I would avoid using `flash.now` in any views. When I said:

> for example by using `flash.now[:warning] = t("layouts.osm_offline")` **in the controller**

... I was thinking about changing the `site#offline` method (in SiteController) something like this:

```diff
-  def offline; end
+  def offline
+    flash.now[:warning] = if Settings.status == "database_offline"
+                            t("layouts.osm_offline")
+                          else
+                            t("layouts.osm_read_only")
+                          end
+  end
```
At this point, the view can be empty, and this will work. Having an empty view is a bit strange, so I would either think of something to put in there (for example, further explanation, but I'm not sure if that is useful) or else remove the view entirely and be explicit in the controller that there's nothing else to render. [I found some examples of how to do this](https://stackoverflow.com/a/45581659/105451), e.g.

```ruby
def offline
  flash.now[...]
  ...
  render :html => nil, :layout => true
end
```

Reply:
I did it but it broke that test:
```
  # Test the offline page
  def test_offline
    get offline_path
    assert_response :success
    assert_template "offline"
  end
```
Do I have to delete the test, since I've deleted `offline.html.erb`?

Reply:
Keep the test, it's still required. The final assertion no longer makes sense and needs to be adjusted - I would change it to no longer asserting that a particular template is used (since we aren't using a view), but to change it to ensure that the layout is still rendered, since we need the layout to make sure the flash message is displayed. 

There are examples of asserting the layout elsewhere in the test directory.

Reply:
I've updated the test and it seems to be working.